### DISCUSSION TITLE
Add a video feedback option rightnow.

### DISCUSSION THREAD

**Discussion by Wowtones** [ntsc-rs/ntsc-rs#173] [Ideas]
I want a feature that allows me to apply the effect multiple times without actually ever exportinh it.

**Comment by Degamisu** [comment]
Taken from #153:

> VHS generation loss was mentioned a long time ago and @valadaptive said something about it not being planned. I can't find the original issue, but it said something along the lines of:
> 
> > This can already be done by just rendering the video and putting it back into the effect
> 
> While this was a good idea, it can easily be replicated by adjusting the bandwidth.
> 
> Hope this helps!
> 
> Note
> 
> All the suggestions can go in the [issues tab](https://github.com/valadaptive/ntsc-rs/issues) as the `discussions` tab is not really read often!

**Reply by Wowtones** [reply]
it cannot be replicated by simply adjusting the bandwidth and i said without exporting it.

**Reply by Degamisu** [reply]
Well, the generation loss can significantly lower the bandwith, hence creating that signature look. If you have the right settings, it will work.

**Reply by Wowtones** [reply]
but i didnt mean simulating generation loss, i meant applying the effect multiple times without exporting.

**Reply by Degamisu** [reply]
I see. But applying the effect multiple times _is_ what generation loss is in my opinion. Here's a good video I've found a long time ago that explains this. You've probably watched [this video](https://www.youtube.com/watch?v=nqy_hYDI0As) before.

Now that I think about it, applying it many times in _real-time_ would significantly reduce performance, and most likely crash the GPU driver or CPU. It would be a pretty dangerous decision that already has a much safer alternative.

**Reply by Wowtones** [reply]
sorry but i still want it

**Reply by jbibel** [reply]
I've been thinking about this discussion for awhile so I tried a little experiment a few weeks ago messing with NTSC-RS in Davinci Resolve to simulate feedback/generation loss. In Davinci Resolve you can construct effects in a node editor so [I did a test](https://www.youtube.com/watch?v=7wtQu8eRRDU) where I had multiple instances of NTSC-RS running into one after the other. This effectively achieves applying the effect multiple times without have to export. Here's a screen shot of how I set it up in this test:
<img width="1050" alt="Screenshot of the Node editor and Keyframe windows of Davinci Resolve. The Node editor shows a MediaIn node going into an NTSC-RS node and then into another four NTSC_RS nodes one after the other and then finally into a MediaOut node. The Keyframe editor shows how each instance of NTSC-RS is keyframed to cascade into each other to simulate feedback/generation loss." src="https://github.com/user-attachments/assets/9c07eae9-84f3-4829-955d-2f75ab9df8fb">

The downside to this process as @Degamisu mention earlier is that it is very computationally intensive when viewing it in real time in Resolve. I'm using a MacBook Air with an M2 processor and 16gb of ram and with five instances open playback was very choppy. I also tried this out with a small project so I don't know how stable this process would be with a larger one. Maybe this isn't what you were looking for originally but I think there are some interesting possibilities with this.

**Reply by Degamisu** [reply]
>I've been thinking about this discussion for awhile so I tried a little experiment a few weeks ago messing with NTSC-RS in Davinci Resolve to simulate feedback/generation loss.

Well done! It's definitely easier to either export the video and put it back into `ntsc-rs`, or adjust the bandwidth.

---
@Wowtones :
> sorry but i still want it

>[!NOTE]
>**TLDR**: Generation loss is typically the result of a loss of bandwidth from the tape's inability to retain HF signals. Adjust the `bandwidth` slider to achieve this effect!

As I've said before, adjusting the bandwidth will achieve this effect. I want to clarify what I meant. `Generation loss` is the same thing as `Signal loss`. This signal degeneration significantly reduces the bandwidth of the tape since there's always a little bit of HF signal loss. The thicker the tape, the more HF is muted. On an oscilloscope, the luminance of a typical `NTSC` signal goes into the HF range, which is lost just like any tape (Even audio cassettes!)

Poorly adjusted distribution amplifiers and mismatched impedances can also cause generation loss!
